routers/userRoutes.js

```diff
@@ -0,0 +1,71 @@
+import { BadRequestError } from "../expressError.js";
+import {
+  userRegisterSchema,
+  userVerifySchema,
+} from "../schemas/userSchemas.js";
+import express from "express";
+import User from "../models/userModel.js";
+import sendEmailRegistration from "../utils/email.js";
+import { validateRequest } from "../middleware/validationMiddleware.js";
+
+const router = new express.Router();
+
+/** POST /user/register: {user} => email OTP (one-time pw)
+ *
+ * User must include {firstName, lastName, email, password}
+ *
+ * Returns 200 status code and sends email with OTP link.
+ *
+ * Authorization required: none */
+
+router.post(
+  "/register",
+  validateRequest([{ schema: userRegisterSchema, reqBody: true }]),
+  async function (req, res, next) {
+    try {
+      const newUser = await User.registerAccount({ ...req.body });
+      const jsonResponse = await sendEmailRegistration(newUser);
+      return res.status(200).json(jsonResponse);
+    } catch (error) {
+      // console.log("error from POST user/register");
+      // console.log(error);
+      //PEER Lawrence, below is my new duplicate constraint error handling where I'm catching the DB-level error to send the custom 'BadRequestError'. Is this okay?
+      if (error.code === "23505" && error.constraint === "users_email_key") {
```

Comment:
Yeah, sending a custom-friendly message is the best practice, so you don't give a hint to malicious users

Reply:
Are there any better methods to catch the error.code from the DB to ensure that the error was caused by a duplicate constraint error?